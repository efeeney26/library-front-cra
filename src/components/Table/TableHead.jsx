import React from 'react'

const TableHead = (props) => {
  return (
    <tr>
      <th>Wow</th>
    </tr>
  )
}

export default TableHead
