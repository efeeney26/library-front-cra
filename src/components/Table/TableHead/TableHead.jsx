import React from 'react'
import PropTypes from 'prop-types'

import styles from './TableHead.module.css'

const TableHead = ({ data }) => (
  <tr>
    {data.map((item, i) => (
      <th
        // добавить nanoid
        key={i}
        className={styles.head}
      >
        {item}
      </th>
    ))}
  </tr>
)

TableHead.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string)
}

TableHead.defaultProps = {
  data: []
}

export default TableHead
