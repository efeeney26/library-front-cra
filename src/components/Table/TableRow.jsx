import React from 'react'

const TableRow = (props) => {
  return (
    <tr>
      <td>1</td>
      <td>2</td>
    </tr>
  )
}

export default TableRow
