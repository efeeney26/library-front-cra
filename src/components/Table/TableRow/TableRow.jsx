import React from 'react'
import PropTypes from 'prop-types'

import styles from './TableRow.module.css'

const TableRow = (props) => {
  const { data } = props
  return (
    data.map((dataItem) => (
      <tr key={dataItem}>
        {dataItem.map((valueItem) => (
          <td
            key={valueItem}
            className={styles.cell}
          >{valueItem}
          </td>
        ))}
      </tr>
    ))
  )
}

TableRow.propTypes = {
  data: PropTypes.arrayOf(PropTypes.array).isRequired
}

export default TableRow
