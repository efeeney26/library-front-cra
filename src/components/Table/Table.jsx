import React from 'react'
import PropTypes from 'prop-types'
import TableRow from './TableRow'
import TableHead from './TableHead'

const Table = (props) => {
  return (
    <table>
      <thead>
        <TableHead />
      </thead>
      <tbody>
        <TableRow />
      </tbody>
    </table>
  )
}

Table.propTypes = {
  children: PropTypes.node
}

Table.defaultProps = {
  children: null
}

export default Table
