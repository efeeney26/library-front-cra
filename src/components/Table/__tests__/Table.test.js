import React from 'react'
import Table from '../Table'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import TableHead from '../TableHead'
import TableRow from '../TableRow'

test('Table snapshot', () => {
  const component = renderer.create(<Table />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('Table', () => {
  const wrapper = shallow(<Table />)
  expect(wrapper.find(TableHead).length).toBe(1)
  expect(wrapper.find(TableRow).length).toBe(1)
})
