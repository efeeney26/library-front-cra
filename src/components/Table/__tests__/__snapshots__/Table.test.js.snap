// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Wow
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        2
      </td>
    </tr>
  </tbody>
</table>
`;
